File: solutions/03 annoying blobs/cs/AnnoyingBlobsTest/DataBlobTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace AnnoyingBlobs
{
    [TestClass]
    public class DataBlobTest
    {
        [TestInitialize]
        public void SetUp()
        {
        }

        [TestCleanup]
        public void TearDown()
        {
        }

        [TestMethod]
        public void TestDataInitiallyEmpty()
        {
            DataBlob blob = DataBlob.GetBlob();

            Assert.IsTrue(blob.IsEmpty());
        }

    }
}

File: exercises/03 annoying blobs/cpp/AnnoyingBlobs/AnnoyingBlobs tests/DataBlobTest.cpp
#include "gtest/gtest.h"
#include "gmock/gmock.h"
#include "DataBlob.h"

class DataBlobTest : public ::testing::Test
{
};

TEST_F(DataBlobTest, testDataInitiallyEmpty)
{
	DataBlob& blob = DataBlob::getBlob();
	EXPECT_TRUE(blob.isEmpty());
}